www/source/docs/run-packages-binding.html.md

```diff
@@ -4,34 +4,56 @@ title: Running packages with runtime binding
 
-For example, you might have a web application `foo` that depends on the value of the leader of a database service group. Rather than hardcoding the name of the service group in `foo`'s plan, which would limit its portability, you can _bind_ the name `database`, for example, to the service group name `postgresql.production` in a production environment, and `postgresql.development` in a development environment.
+For example, you might have a web application `app-server` that depends on the value of the leader of a database service group. Rather than hardcoding the name of the service group or package identifier in `app-servers`'s plan, which would limit its portability, you can _bind_ the name `database`, for example, to the `default` service group running PostgreSQL. If you have multiple service groups for PostgreSQL - perhaps you have a production and development environment - you could bind `database` to `postgresql.production` or `postgresql.development`. If `app-server` supports multiple different database backends you could even bind `database` to another, such as `redis.default` or `mysql.default`.
 
-## Example
+## Producer Contract
 
-For an example look at the [haproxy](https://github.com/habitat-sh/core-plans/blob/master/haproxy/config/haproxy.conf) core-plan. These lines in the [haproxy.conf](https://github.com/habitat-sh/core-plans/blob/master/haproxy/config/haproxy.conf#L18-L23) illustrate how to reference a service binding. When a binding is enabled, the config will be rendered with a backend for each healthy instance in the service group.
+The producer defines their contract by "exporting" configuration publicly to consumers. This is done by setting keys in the `pkg_exports` associative array defined in your package's `plan.sh`. For example, a database server named `amnesia` might define the exports:
+
+      pkg_exports=(
+        [port]=network.port
+        [ssl-port]=network.ssl.port
+      )
+
+This will export the value of `network.port` and `transport.ssl.port` defined in it's `default.toml` publicly as `port` and `ssl-port` respectively. All `pkg_exports` must define a default value in `default.toml` but their values may change at runtime by an operator configuring the service group. If this happens, the consumer will be notified that their producer's configuration has changed. We'll see how to leverage this on the consumer in the sections below.
```

Comment:
Do you mean `network.ssl.port` in the description, or do you want to change the code example?